### ISSUE TITLE
Webpacker ^6.0.0-beta.6: TypeError: Invalid PostCSS Plugin found at: plugins[0]

### ISSUE THREAD

**Issue by anitagraham** [rails/webpacker#2965]
I wrote all the below up, then had a brain wave and reduced `postcss.config.js` to
```
module.exports={}
```
And now it all works.  But where did `postcss.config.js` come from? Should it have worked?

Rails 6.0.3.5
Webpacker ^6.0.0-beta.6

I've detailed the changes I've made to the standard configuration at the end of the issue.

The webpacker run (`yarn run dev`) nearly completes until...

```
ERROR in ./app/javascript/src/stylesheets/pages/application.scss
Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
ModuleBuildError: Module build failed (from ../node_modules/postcss-loader/lib/index.js):
TypeError: Invalid PostCSS Plugin found at: plugins[0]

(@/private/var/www/rummy/postcss.config.js)
    at /private/var/www/node_modules/postcss-load-config/src/plugins.js:78:15
    at Array.forEach (<anonymous>)
    at plugins (/private/var/www/node_modules/postcss-load-config/src/plugins.js:60:13)
    at processResult (/private/var/www/node_modules/postcss-load-config/src/index.js:33:14)
    at /private/var/www/node_modules/postcss-load-config/src/index.js:94:14
    at processResult (/private/var/www/rummy/node_modules/webpack/lib/NormalModule.js:674:19)
    at /private/var/www/rummy/node_modules/webpack/lib/NormalModule.js:773:5
    at /private/var/www/rummy/node_modules/loader-runner/lib/LoaderRunner.js:399:11
    at /private/var/www/rummy/node_modules/loader-runner/lib/LoaderRunner.js:251:18
    at context.callback (/private/var/www/rummy/node_modules/loader-runner/lib/LoaderRunner.js:124:13)
    at /private/var/www/node_modules/postcss-loader/lib/index.js:194:71
 @ ./app/javascript/src/application.js 1:0-44
 @ ./app/javascript/packs/application.js 7:0-31
```

**Note:** Earlier Sass/css errors have been resolved so I am assuming the Sass => css step has succeeded.
**Note:** I was also getting this error when I was using relative paths, rather than the aliases which is what most of local configuration is about.

```
// postcss.config.js
module.exports = {
  plugins: [
    require('postcss-import'),
    require('postcss-flexbugs-fixes'),
    require('postcss-preset-env')({
      autoprefixer: {
        flexbox: 'no-2009'
      },
      stage: 3
    })
  ]
}
```
I have added a couple of console.log statements to `postcss-load-config/src/plugins`, which tell me that

config.plugins is
```
[
  [Function: AtImport] { postcss: true },
  [Function (anonymous)] { postcss: true },
  [Function (anonymous)] {
    postcssPlugin: 'postcss-preset-env',
    postcssVersion: '7.0.35'
  }
]
```
which seems to reflect `postcss.plugins.js`, which is living at the root of the project.
(If this is part of the webpacker default config shouldn't it be in `config/webpacker`? ).

Earlier (before reinstalling everything) I tried to remove `postcss` from the project - as the documentation implies that it is an optional feature you can choose to install, but it seems to be a requirement.

___
Changes to configuration:
```javascript
// config/webpack/base.js
const { webpackConfig, merge } = require('@rails/webpacker')
const custom = require('./custom')
module.exports = merge(webpackConfig, custom)
```

```javascript
// config/webpack/custom.js
// defining path aliases 
const path = require('path');
const { config } = require('@rails/webpacker')

const jsRoot = config.source_path;
const Packs = config.source_entry_path
const Public = config.public_root_path

const Source =  path.resolve(jsRoot, 'src');
const railsRoot = path.resolve(jsRoot, '..');
const SourceBranch = name => path.resolve(Source, name)

const custom = {
  resolve: {
    alias: {
      Public:     config.public_root_path,
      Source:     Source,
      Components: SourceBranch('components'),
      Images:     SourceBranch('images'),
      Settings:   SourceBranch('settings'),
      Stylesheets: SourceBranch('stylesheets'),
      Utilities:  SourceBranch('utilities')
    },
    extensions: ['.css']
  }
module.exports custom
```

**Comment by mcmire** [comment]
It's a little hard to follow your steps here, so if you could reproduce what you were doing then I'm sure that would be helpful. However, one thing I am noticing in some of your output is that you may be using PostCSS 7. PostCSS 8 is the latest version, so perhaps it might work to upgrade that in your `package.json`?

**Comment by anitagraham** [comment]
Thanks @mcmire.  In the end I gave up on `webpacker` again, and have gone back to pure `webpack`.
